source/upgrade/upgrading-mattermost-server.rst

```diff
@@ -1,44 +1,116 @@
+      See the `Mattermost v6.0 DB Schema Migrations Analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#online-migration-mysql>`__ documentation for a sample execution and additional caveats.
 
-For `High Availability <https://docs.mattermost.com/scale/high-availability-cluster.html>`__ deployments, we only support one minor version difference between  server versions when performing a rolling upgrade. For example v5.27.1 + v5.27.2 or v5.26.4 + v5.27.1 is supported, whereas v5.25.5 + v5.27.0 is not supported. Running two different versions of Mattermost in your cluster should not be done outside of an upgrade scenario. In some upgrade scenarios, it won't be possible to run different versions, and such cases are clearly noted in the :doc:`important-upgrade-notes` product documentation.
+   .. tab:: PostgreSQL
+
+      The following query executed during the migration process will have a significant impact on database CPU usage and write operation restrictions for the duration of the query:
+
+      ``ALTER TABLE posts ALTER COLUMN props TYPE jsonb USING props::jsonb;`` (~ 11 minutes)
+
+      For a complete breakdown of PostgreSQL queries, as well as their impact and duration, see the `Mattermost v6.0 DB Schema Migrations Analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#postgresql-1>`__.
+
+Upgrading from Releases Older than v5.35
+----------------------------------------
+
+Customers upgrading from a release older than Mattermost v5.35 should expect downtime when upgrading to v6.0, due to the introduction of backend database architecture introduced in v5.35. See the `Mattermost Changelog <https://docs.mattermost.com/install/self-managed-changelog.html?highlight=changelog#id39>`__ documentation for details.
+
+If you're upgrading from a version prior to v5.0, be sure to also modify your service file to work with the binary changes introduced with v5.0. Your execution directory should point to the Mattermost base directory (i.e. ``/opt/mattermost``), and your binary should point to the ``mattermost`` binary (i.e. ``/opt/mattermost/bin/mattermost``).
+
+Upgrading High Availability Deployments
+---------------------------------------
+
+In `High Availability <https://docs.mattermost.com/scale/high-availability-cluster.html>`__ environments, you should expect to schedule downtime for the upgrade to v6.0. Based on your database size and setup, the migration to v6.0 can take a significant amount of time, and may even lock the tables for posts which will prevent you from posting or receiving messages until the migration is complete.
+
+Ensure you review the `High Availability Cluster Upgrade Guide <https://docs.mattermost.com/scale/high-availability-cluster.html#upgrade-guide>`__, as well as the :doc:`important-upgrade-notes` to make sure you're aware of any actions you need to take before or after upgrading from your particular version.
```

Comment:
Will the upgrade guide here need updating too?

Reply:
Great question, @wiersgallak. @streamer45 & @spirosoik - Are you open to reviewing the High Availability page for v6.0 impact, please?

Reply:
Given 6.0 introduces changes that require some downtime on both HA and non-HA installations I am thinking there isn't any particular info that would specifically apply to HA. Obviously for such installation I'd be strongly recommending doing an online migration of the Posts table to minimize the downtime as much as possible.

@agnivade Anything of note we may be missing here?

Reply:
Just the note that a 5.x node cannot run with 6.0 node in a cluster, which is already pointed out below. So all good :+1: